### ISSUE TITLE
Why choose libraries based on riscv64-* or riscv32-**instead of xlen?

### ISSUE THREAD

**Issue by majin2020** [kraj/musl#2]
I found that when I use the compiler riscv64, no matter ```-march = rv32gc``` or ```-march = rv64gc```, ```typedef signed long int64_t;``` always appears. Obviously, when ```-march = rv32gc```, it is correct to ```typedef the signed long long int64_t;```.

**Comment by kraj** [comment]
yes, we can correct that, Care to send a patch ? RV32 port is still not upstream.
